### ISSUE TITLE
Error When Installing OctoFarm

### ISSUE THREAD

**Issue by RonenGru** [OctoFarm/OctoFarm#983]
### OctoFarm Version

1.2

### OctoPrint Version(s)

1.8

### pm2 or docker?

pm2

### Operating System

Linux Mint (Ubuntu)

### Location of the problem

Server

### Description of the problem

```markdown
I tried to install OctoFarm several times with the same problem.  I have had OctoFarm running without issue previously but with the older version of OctoFarm.

I installed a fresh installation of Linux Mint on the computer and followed all the steps at https://docs.octofarm.net/installation/install-linux-ubuntu.html.  Everything installed as expected (using terminal & SSH).  When running npm start I get the following error(s):

npm ERR! fsevents not accessible from chokidar

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/rg/.npm/_logs/2022-05-22T22_07_37_462Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! octofarm@1.2.2 prestart: `npm ci`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the octofarm@1.2.2 prestart script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/rg/.npm/_logs/2022-05-22T22_07_37_547Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! octofarm-entry@1.0.1 start: `cd server && npm run start`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the octofarm-entry@1.0.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/rg/.npm/_logs/2022-05-22T22_07_37_615Z-debug.log

I tried reinstalling a number of times but with the same result.  Not sure what is going on exactly or how to fix.

Here are the error logs:

info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'ci' ]
2 info using npm@6.14.17
3 info using node@v14.19.3
4 verbose npm-session 162fdccd7f1cc455
5 info prepare initializing installer
6 verbose prepare starting workers
7 verbose prepare installation prefix: /home/rg/OctoFarm/server
8 verbose prepare using package-lock.json
9 verbose checkLock verifying package-lock data
10 verbose teardown shutting down workers.
11 info teardown Done in 0.001s
12 verbose stack Error: fsevents not accessible from chokidar
12 verbose stack     at reqAddr (/usr/lib/node_modules/npm/node_modules/npm-logical-tree/index.js:159:17)
12 verbose stack     at /usr/lib/node_modules/npm/node_modules/npm-logical-tree/index.js:129:22
12 verbose stack     at Array.forEach (<anonymous>)
12 verbose stack     at addChild (/usr/lib/node_modules/npm/node_modules/npm-logical-tree/index.js:128:40)
12 verbose stack     at /usr/lib/node_modules/npm/node_modules/npm-logical-tree/index.js:133:7
12 verbose stack     at Array.forEach (<anonymous>)
12 verbose stack     at addChild (/usr/lib/node_modules/npm/node_modules/npm-logical-tree/index.js:128:40)
12 verbose stack     at /usr/lib/node_modules/npm/node_modules/npm-logical-tree/index.js:113:5
12 verbose stack     at Array.forEach (<anonymous>)
12 verbose stack     at lockTree (/usr/lib/node_modules/npm/node_modules/npm-logical-tree/index.js:107:5)
12 verbose stack     at /usr/lib/node_modules/npm/node_modules/libcipm/index.js:144:21
13 verbose cwd /home/rg/OctoFarm/server
14 verbose Linux 5.4.0-110-generic
15 verbose argv "/usr/bin/node" "/usr/bin/npm" "ci"
16 verbose node v14.19.3
17 verbose npm  v6.14.17
18 error fsevents not accessible from chokidar
19 verbose exit [ 1, true ]

 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'start' ]
2 info using npm@6.14.17
3 info using node@v14.19.3
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle octofarm@1.2.2~prestart: octofarm@1.2.2
6 verbose lifecycle octofarm@1.2.2~prestart: unsafe-perm in lifecycle true
7 verbose lifecycle octofarm@1.2.2~prestart: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/rg/OctoFarm/server/node_modules/.bin:/usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/rg/O>
8 verbose lifecycle octofarm@1.2.2~prestart: CWD: /home/rg/OctoFarm/server
9 silly lifecycle octofarm@1.2.2~prestart: Args: [ '-c', 'npm ci' ]
10 silly lifecycle octofarm@1.2.2~prestart: Returned: code: 1  signal: null
11 info lifecycle octofarm@1.2.2~prestart: Failed to exec prestart script
12 verbose stack Error: octofarm@1.2.2 prestart: `npm ci`
12 verbose stack Exit status 1
12 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
12 verbose stack     at EventEmitter.emit (events.js:400:28)
12 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
12 verbose stack     at ChildProcess.emit (events.js:400:28)
12 verbose stack     at maybeClose (internal/child_process.js:1088:16)
12 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:296:5)
13 verbose pkgid octofarm@1.2.2
14 verbose cwd /home/rg/OctoFarm/server
15 verbose Linux 5.4.0-110-generic
16 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "start"
17 verbose node v14.19.3
18 verbose npm  v6.14.17
19 error code ELIFECYCLE
20 error errno 1
21 error octofarm@1.2.2 prestart: `npm ci`
21 error Exit status 1
22 error Failed at the octofarm@1.2.2 prestart script.
22 error This is probably not a problem with npm. There is likely additional logging output above.
23 verbose exit [ 1, true ]

 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'start' ]
2 info using npm@6.14.17
3 info using node@v14.19.3
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle octofarm-entry@1.0.1~prestart: octofarm-entry@1.0.1
6 info lifecycle octofarm-entry@1.0.1~start: octofarm-entry@1.0.1
7 verbose lifecycle octofarm-entry@1.0.1~start: unsafe-perm in lifecycle true
8 verbose lifecycle octofarm-entry@1.0.1~start: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/rg/OctoFarm/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/loc>
9 verbose lifecycle octofarm-entry@1.0.1~start: CWD: /home/rg/OctoFarm
10 silly lifecycle octofarm-entry@1.0.1~start: Args: [ '-c', 'cd server && npm run start' ]
11 silly lifecycle octofarm-entry@1.0.1~start: Returned: code: 1  signal: null
12 info lifecycle octofarm-entry@1.0.1~start: Failed to exec start script
13 verbose stack Error: octofarm-entry@1.0.1 start: `cd server && npm run start`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:400:28)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:400:28)
13 verbose stack     at maybeClose (internal/child_process.js:1088:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:296:5)
14 verbose pkgid octofarm-entry@1.0.1
15 verbose cwd /home/rg/OctoFarm
16 verbose Linux 5.4.0-110-generic
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "start"
18 verbose node v14.19.3
19 verbose npm  v6.14.17
20 error code ELIFECYCLE
21 error errno 1
22 error octofarm-entry@1.0.1 start: `cd server && npm run start`
22 error Exit status 1
23 error Failed at the octofarm-entry@1.0.1 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

Any help would be appreciated.  Ronen

After a bit of experimenting I was able to get npm to work by updating nodejs to version 17.9.0 using the following commands:

sudo apt remove --purge nodejs npm
sudo apt clean
sudo apt autoclean
sudo apt install -f
sudo apt autoremove
sudo apt install curl
cd ~
curl -fsSL https://deb.nodesource.com/setup_17.x | sudo -E bash -
sudo apt-get install -y nodejs
curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
sudo apt update
sudo apt install yarn
node -v && npm -v
```

### Steps taken to reproduce

Problem always occurs with Linux Mint 20.3 using the online installation procedure found at https://docs.octofarm.net/installation/install-linux-ubuntu.html

### What did you expect to happen

I expect OctoFarm to run and to be accessible via the browser.

### To avoid the problem I had to

I was unable to avoid the problem after multiple attempts to install.

**Comment by NotExpectedYet** [comment]
Update your npm
`sudo npm install -g npm@latests` is all that was needed

**Comment by NotExpectedYet** [comment]
I'll add it to the instruction set

**Comment by NotExpectedYet** [comment]
I don't test or support nodejs 17 just so your aware. Only LTS versions.

**Comment by NotExpectedYet** [comment]
Fixed the docs, closing.

**Comment by RonenGru** [comment]
Thanks!  Reinstalled with the fix.  All is working.  Excellent software.  Ronen

    On Monday, May 23, 2022, 11:42:57 p.m. EDT, James Mackay ***@***.***> wrote:  
 
 


Closed #983 as completed.

**Comment by NotExpectedYet** [comment]
> Thanks!  Reinstalled with the fix.  All is working.  Excellent software.  Ronen On Monday, May 23, 2022, 11:42:57 p.m. EDT, James Mackay ***@***.***> wrote: Closed #983 as completed. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.Message ID: ***@***.***>

Thanks for the praise!
